components/function-controller/internal/controllers/serverless/function_reconcile_matchers_test.go

```diff
@@ -0,0 +1,91 @@
+package serverless
```

Comment:
I noticed most test files use the `serverless` package instead of `serverless_test`, cluttering the scope. Maybe we could investigate how much effort would it require to move the unit tests to the `serverless_test` package? Not necessarily in this PR.